marcello3d/trimerge-sync
# Fix IndexedDB contention issue

Previously, the transaction to write commits could be interrupted by attempting to fetch the localStoreId before the promise had resolved because of the way IndexedDb transactions work.

Since local id is never expected to change, this PR reads that data on connection to the DB and returns that alongside the database object so that we don't need to interrupt our transaction by awaiting the result of the LocalStoreId promise.

## Tests
 - Confirm this fixes the issue
 - Confirm that local store ids are set as expected
 - Confirm that commits are sent correctly to the server and we don't see any problem sending merge commits to the server
